flang/runtime/command.cpp

```diff
@@ -239,4 +241,22 @@ std::int32_t RTNAME(GetEnvVariable)(const Descriptor &name,
   return StatOk;
 }
 
+std::int32_t RTNAME(GetCwd)(
+    const Descriptor &cwd, const char *sourceFile, int line) {
+  Terminator terminator{sourceFile, line};
+
+  RUNTIME_CHECK(terminator, IsValidCharDescriptor(&cwd));
+
+  char *buf = getcwd(nullptr, 0);
```

Comment:
POSIX does not require that `getcwd(nullptr, 0)` allocates memory with `malloc`. It would be standards compliant to just return nullptr here.

My glibc man page says that it does implement the memory allocation as an extension to the POSIX standard. If this extension is present on all implementations where people are using flang in practice (MacOS, *BSD, Windows, AIX(?)) then this is okay, otherwise we cannot rely on this behavior.

See https://pubs.opengroup.org/onlinepubs/009695399/functions/getcwd.html

Reply:
In the least, this assumption should be documented as a comment.

Reply:
It looks like doing that (`getcwd(nullptr, ...)`) on AIX is not recommended.
https://www.ibm.com/docs/en/aix/7.3?topic=g-getcwd-subroutine

Reply:
It is indeed unspecified in some standards. Thank you for pointing out!